gcloud-java-examples/src/main/java/com/google/gcloud/examples/DnsExample.java

```diff
@@ -0,0 +1,516 @@
+      String dnsName = args[1];
+      String description = args[2];
+      ZoneInfo zoneInfo = ZoneInfo.builder(zoneName)
+          .dnsName(dnsName)
+          .description(description)
+          .build();
+      Zone zone = dns.create(zoneInfo);
+      System.out.printf("Successfully created zone with name %s which was assigned ID %s.%n",
+          zone.name(), zone.id());
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name> <dns_name> <description>";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 3;
+    }
+  }
+
+  private static class ListZonesAction implements DnsAction {
+
+    /**
+     * Lists all zones within the project.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      Iterator<Zone> zoneIterator = dns.listZones().iterateAll();
+      if (zoneIterator.hasNext()) {
+        System.out.println("The project contains the following zones:");
+        System.out.println("Name\tID\tDNS Name\tCreated\tDesription");
+        while (zoneIterator.hasNext()) {
+          Zone zone = zoneIterator.next();
+          System.out.printf("%s\t%s\t%s\t%s\t%s%n", zone.name(), zone.id(), zone.dnsName(),
+              zone.creationTimeMillis(), zone.description());
+        }
+      } else {
+        System.out.println("Project contains no zones.");
+      }
+    }
+
+    @Override
+    public String params() {
+      return "";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 0;
+    }
+  }
+
+  private static class GetZoneAction implements DnsAction {
+
+    /**
+     * Gets details about a zone with the given name.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      Zone zone = dns.getZone(zoneName);
+      if (zone == null) {
+        System.out.printf("No zone with name '%s' exists.%n", zoneName);
+      } else {
+        System.out.printf("Name: %s%n", zone.name());
+        System.out.printf("ID: %s%n", zone.id());
+        System.out.printf("Description: %s%n", zone.description());
+        System.out.printf("Created: %s%n", zone.creationTimeMillis());
+        System.out.printf("Name servers: %s%n", Joiner.on(",").join(zone.nameServers()));
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name>";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 1;
+    }
+  }
+
+  private static class DeleteZoneAction implements DnsAction {
+
+    /**
+     * Deletes a zone with the given name.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      boolean deleted = dns.delete(zoneName);
+      if (deleted) {
+        System.out.printf("Zone %s was deleted.%n", zoneName);
+      } else {
+        System.out.printf("Zone %s was NOT deleted. It probably does not exist.%n", zoneName);
```

Comment:
In what circumstances could the boolean be false but the zone does/did exist? I ask because it'd be nice to avoid the word "probably" in this print statement.